### ISSUE TITLE
Specify a camera's pivot location

### ISSUE THREAD

**Issue by zachernuk** [immersive-web/model-element#72]
At TPAC we discussed the necessity to point the view into a `<model>` element at specific locations within it, e.g. to focus on specific features of a car. 

We _also_ agreed that for both reasons of complexity for the author and compatibility across platforms, that supplying view information like a 4x4 projection matrix  were not appropriate. Instead, to follow something closer to the style indicated with the existing proposed JS syntax, i.e `setCamera({pitch:10, yaw:10, scale:1})`.

My suggestion is to specify an `{x,y,z}` object as an optional `origin` or `pivot` parameter on that setCamera call:
```javascript
shelf.setCamera( { 
  pitch:0, 
  yaw:20, 
  scale:1, 
  pivot: {x:0, y:1, z:0}
} );
``` 
 and in fact for the other parameters to be made optional as well, with the presumption that no supplied value implies to retain the same one.

**Comment by cabanier** [comment]
Can you explain what `shelf` represents? Are you defining a new API on the model element?

**Comment by zachernuk** [comment]
aha - no! I have been using a 3D model of a shelf / shelf configurator as a sample for doing this (and the IBL example), so I was naming the hypothetical model based on what it is doing. I've updated the name to be `myShelf` to reflect that it's a user object rather than an API name.

**Comment by DR-xR** [comment]
Are the units for rotation in radians or degrees? I suspect from the provided values that it is degrees, but JS deals with radians. There is the potential issue of using one unit for the API and a different one for the HTML because it is more convenient for the user and environment.

It is also necessary to define the origin (up & forward) and location. Location is necessary because how far you look up (for example) depends on how far away from the object you are looking at. If that is included in `pivot`, then I am confused. If that is to be the location of the camera (relative to the local origin), then why not use `location`.

**Comment by zachernuk** [comment]
After a lot of exploration I think it would be better to pursue a metaphor where the root element  (possibly even denoted as `model.rootElement`) exposes the transform it's using. 

This avoids the notion of a "camera" as an obligatory feature at all, which would help clarify its function in spatial/stereoscopic environments. 

And while I think we can and should pursue a V1 specification (and implementation) that doesn't reach into a scene graph, exposing the root node will give us a reasonable pattern to follow once we have the appetite to do so. 

/facetoface

**Comment by zachernuk** [comment]
from discussions at TPAC I think the `DOMMatrix`-based transformation seems adequate - we could try to investigate [XRRigidTransform](https://developer.mozilla.org/en-US/docs/Web/API/XRRigidTransform) but I do suspect it has _more_ information in it than we actually want. 

_Another_ thing we get out of using DOMMatrix is that we could aim to specify the (entity|view|model)transform within _CSS_, which opens the door to styling and transitions, and leveraging CSS animation on the properties. I find that very exciting. @toji, is there anything that we'd be missing out on from `DOMMatrix`?

**Comment by toji** [comment]
I don't think so. Though to be a bit pedantic: `XRRigidTransform` is actually _more limited_ than `DOMMatrix` in terms of the transforms that it can represent.

The reason it was discussed at all is because there was some concern about supporting skew, which `DOMMatrix` allows and `XRRigidTransform` would avoid. However `XRRigidTransform` also doesn't allow for scale, which seems fairly important for this use case. As such even though we'll need to figure out how to handle skew it does seem like `DOMMatrix` is the better option here. (In addition to the CSS capabilities mentioned.)

**Comment by zachernuk** [comment]
/agenda
Do we want to codify that this happens both _with_ `DOMMatrix` and _in_ CSS? It's not essential to do both, but it does mean things like `transition-duration` and CSS Animations have a way to being used.

**Comment by mkeblx** [comment]
Since it seems like we may not desire to support skew, and thus all possible matrix values, we would need to define what exactly a valid and invalid `DOMMatrix` would be.

Also we need to define the behavior of setting an invalid `DOMMatrix` to a `model`'s `entityTransform`.
